the-collab-lab/tcl-79-smart-shopping-list
# Mm feature/navbar layout

## Description
This PR introduces a fully responsive navigation system for both desktop and tablet/mobile devices.
An active link class has been applied to highlight the current page the user is on, along with hover states for the links and icons.
Additionally, a light and dark mode toggle has been implemented, allowing users to switch between these two themes.
This enhances the user experience by ensuring the navigation is intuitive across devices and provides flexibility in user interface preferences.



## Related Issue
Closes #31 



## Acceptance Criteria

- [x]  Mobile nav bar version is similar to the one we currently have.
- [x]  Desktop nav bar is on the top as Figma design. 
- [x]  Remove manage list page since we moved everything out of it. 



## Type of Changes

`feature`
`enhancement` 

## Updates

### Before

![Screenshot 2024-10-05 at 17 11 02](https://github.com/user-attachments/assets/c1e2a903-799e-44cb-b8ec-ba290354da64)


![Screenshot 2024-10-05 at 16 37 59](https://github.com/user-attachments/assets/c7cd41df-84fb-49a5-9e56-f20f703214c2)
![Screenshot 2024-10-05 at 16 38 27](https://github.com/user-attachments/assets/2e528ea8-cb69-4bcb-8188-46a86c4da883)




### After

#### Favicon

![Screenshot 2024-10-05 at 16 53 12](https://github.com/user-attachments/assets/bd2d9100-8d11-459d-90e4-929b2bcadbde)


#### Desktop Light mode
![Screenshot 2024-10-05 at 16 43 46](https://github.com/user-attachments/assets/2f515efd-0652-4828-a69b-31c05183379b)


![Screenshot 2024-10-05 at 16 46 09](https://github.com/user-attachments/assets/0400d241-0bbb-42c5-8a70-03b82e07d5dd)


#### Desktop Dark mode
![Screenshot 2024-10-05 at 16 45 22](https://github.com/user-attachments/assets/e33ca64a-c271-4edb-9f75-ec89c1f7cbf5)


<img width="1162" alt="Screenshot 2024-10-05 at 16 44 08" src="https://github.com/user-attachments/assets/f048f9ad-5eb6-44d8-9c18-0476b4fce63e">

#### Mobile Light mode
![Screenshot 2024-10-05 at 16 47 07](https://github.com/user-attachments/assets/0cc64960-ded7-40c9-ad82-264a977faee4)
![Screenshot 2024-10-05 at 16 47 18](https://github.com/user-attachments/assets/de0b2134-577e-48f7-b665-968f4db0b8bc)

#### Mobile Dark mode

![Screenshot 2024-10-05 at 16 47 34](https://github.com/user-attachments/assets/f0fa33ec-936e-4319-bd93-843b6693abab)
![Screenshot 2024-10-05 at 16 47 46](https://github.com/user-attachments/assets/c5ebca09-e9c0-4b76-9716-249188cbfce7)




## Testing Steps / QA Criteria

- Do a `git pull` and `git checkout mm-feature/navbar-layout` 
- After switching you can see the responsive navbar displayed. You can switch between desktop and tablet/mobile views to observe the different layouts
- You can also toggle between Light mode and Dark mode to test the theme-switching functionality.
